src/cascadia/TerminalApp/App.cpp

```diff
@@ -713,14 +714,31 @@ namespace winrt::TerminalApp::implementation
     void App::_UpdateTitle(std::shared_ptr<Tab> tab)
     {
         auto newTabTitle = tab->GetFocusedTitle();
+        const auto lastFocusedProfile = tab->GetFocusedProfile().value();
+        const auto* const matchingProfile = _settings->FindProfile(lastFocusedProfile);
+
+        auto tabTitle = matchingProfile->GetTabTitle();
 
-        // TODO #608: If the settings don't want the terminal's text in the
-        // tab, then display something else.
-        tab->SetTabText(newTabTitle);
-        if (_settings->GlobalSettings().GetShowTitleInTitlebar() &&
-            tab->IsFocused())
+        // Checks if tab title has been set in the profile settings and
+        // updates accordingly.
+
+        if (tabTitle.empty())

```

Comment:
Both the `if` and the `else` blocks look like they're fundamentally doing the same thing - could we maybe define a `const auto newActualTitle = tabTitle.empty() ? newTabTitle : tabTitle;` and then just have one 
```c++
            tab->SetTabText(newActualTitle);
            if (_settings->GlobalSettings().GetShowTitleInTitlebar() &&
                tab->IsFocused())
            {
                _titleChangeHandlers(newActualTitle);
            }
```
?